articles/api-auth/tutorials/adoption/password.md

```diff
@@ -0,0 +1,192 @@
+---
+title: OIDC-conformant Resource Owner Password Credentials exchange
+---
+
+# Resource Owner Password Credentials exchange
+
+<%= include('./_about.md') %>
+
+The [Resource Owner Password Credentials exchange](/api-auth/grant/password) is used by highly-trusted clients to provide active authentication.
+Unlike the authorization code and implicit grants, this authentication mechanism does not redirect users to Auth0.
+It authenticates users with a single request, exchanging their password credentials for a token.
+This document describes the differences of this flow between the legacy and OIDC-conformant authentication pipelines.
+
+## Authentication request
+
+<div class="code-picker">
+  <div class="languages-bar">
+    <ul>
+      <li><a href="#request-legacy" data-toggle="tab">Legacy</a></li>
+      <li><a href="#request-oidc" data-toggle="tab">OIDC-conformant</a></li>
+    </ul>
+  </div>
+  <div class="tab-content">
+    <div id="request-legacy" class="tab-pane active">
+      <pre class="text hljs"><code>POST /oauth/ro HTTP 1.1
+Content-Type: application/json
+{
+  "grant_type": "password",
+  "client_id": "123",
+  "username": "alice",
+  "password": "A3ddj3w",
+  "connection": "my-database-connection",
+  "scope": "openid email favorite_color offline_access",
+  "device": "my-device-name"
+}</code></pre>
+    <ul>
+        <li>The <code>device</code> parameter is only needed if <a href="/tokens/refresh-token">requesting a refresh token</a> by passing the <code>offline_access</code> scope.</li>
+    </ul>
+    </div>
+    <div id="request-oidc" class="tab-pane">
+      <pre class="text hljs"><code>POST /oauth/token HTTP 1.1
+Content-Type: application/json
+{
+  "grant_type": "http://auth0.com/oauth/grant-type/password-realm",
```

Comment:
The `/oauth/token` endpoint also supports the standard Password Grant (`grant_type: password`) and not just our enhanced `password-realm`. A customer uses `grant_type: password` they also have to set the `tenant.default_directory` setting to the name of the connection to use for this grant type.

I think `password-realm` is a better apples-to-apples comparison with `/oauth/ro` so it's probably best to keep it as the grant type in the example, but we might want to footnote mention `password` as well.